container/container_unix.go

```diff
@@ -253,29 +253,40 @@ func (container *Container) IpcMounts() []Mount {
 	return mounts
 }
 
-// SecretMount returns the mount for the secret path
-func (container *Container) SecretMount() *Mount {
-	if len(container.SecretReferences) > 0 {
-		return &Mount{
-			Source:      container.SecretMountPath(),
-			Destination: containerSecretMountPath,
+// SecretMounts returns the mount for the secret path
+func (container *Container) SecretMounts() []Mount {
+	var mounts []Mount
+	for _, r := range container.SecretReferences {
+		// secrets are created in the SecretMountPath at a single level
+		// i.e. /var/run/secrets/foo
+		srcPath := filepath.Join(container.SecretMountPath(), filepath.Base(r.File.Name))
+		mounts = append(mounts, Mount{
+			Source:      srcPath,
+			Destination: r.File.Name,
```

Comment:
Doesn't this need to be prefixed with `containerSecretMountPath` if the target is not an absolute path?